tests/commands/envelope_test.py

```diff
@@ -355,3 +356,41 @@ def test_apply_from_user_and_email(self):
 
         self.assertTrue(env.sign)
         self.assertIs(env.sign_key, mock.sentinel.gpg_key)
+
+
+class TestSendCommand(unittest.TestCase):
```

Comment:
This is a testing an asynchronous call, I don't think you can't use `unittest` for this. I think you need to use `twisted.trial.unittest` (I'm still slightly fuzzy on testing twisted, I use asyncio and python3 when I want to do asynchronous programming).

Have a look at the commands/global_test.py tests to see the differences. I think that using `@inlineCallbacks` is much easier than building a `deferred` to test with manually.